### PR TITLE
Fix building without features

### LINKED ISSUE #3684: Can't build test without WITH_XC_YUBIKEY=ON
## Expected Behavior
I can build the tests without yubikey like in version 2.4.3

## Current Behavior
```
FAILED: tests/CMakeFiles/testcli.dir/TestCli.cpp.o 
/usr/ports/pobj/keepassxc-2.5.0/bin/c++  -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_EXCEPTIONS -DQT_STRICT_ITERATORS -DQT_SVG_LIB -DQT_TESTCASE_BUILDDIR=\"/usr/ports/pobj/keepassxc-2.5.0/build-amd64\" -DQT_TESTLIB_LIB -DQT_TEST_LIB -DQT_WIDGETS_LIB -DUSE_READLINE -DWITH_APP_BUNDLE -Itests/testcli_autogen/include -I/usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/tests -Itests -I/usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/src -Itests/../src -isystem /usr/local/include -isystem /usr/local/include/X11/qt5 -isystem /usr/local/include/X11/qt5/QtCore -isystem /usr/local/lib/qt5/./mkspecs/openbsd-clang -isystem /usr/local/include/X11/qt5/QtConcurrent -isystem /usr/local/include/X11/qt5/QtWidgets -isystem /usr/local/include/X11/qt5/QtGui -isystem /usr/X11R6/include -isystem /usr/local/include/X11/qt5/QtTest -isystem /usr/local/include/X11/qt5/QtSvg -isystem /usr/local/include/X11/qt5/QtDBus -isystem /usr/local/include/X11/qt5/QtNetwork -O2 -pipe -g -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Werror=format-security -std=c++11 -fsized-deallocation -DNDEBUG   -fPIC -MD -MT tests/CMakeFiles/testcli.dir/TestCli.cpp.o -MF tests/CMakeFiles/testcli.dir/TestCli.cpp.o.d -o tests/CMakeFiles/testcli.dir/TestCli.cpp.o -c /usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/tests/TestCli.cpp
/usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/tests/TestCli.cpp:1794:10: error: use of undeclared identifier 'YubiKey'
    if (!YubiKey::instance()->init()) {
         ^
/usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/tests/TestCli.cpp:1799:23: error: use of undeclared identifier 'YubiKey'
    bool isBlocking = YubiKey::instance()->checkSlotIsBlocking(2, errorMessage);
                      ^
/usr/ports/pobj/keepassxc-2.5.0/keepassxc-2.5.0/tests/TestCli.cpp:1806:5: error: use of undeclared identifier 'YubiKey'
    YubiKey::instance()->challenge(2, false, challenge, response);
    ^
3 errors generated.
ninja: build stopped: subcommand failed

```

## Steps to Reproduce
Build keepassxc 2.5.0 with -DWITH_GUI_TESTS=ON **without** -DWITH_XC_YUBIKEY=ON

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )

## Debug Info
[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.5.0

Operating system: OpenBSD
CPU architecture: amd64

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ Refactor (significant modification to existing code)
- ✅ New feature (non-breaking change which adds functionality)
- ✅ Breaking change (fix or feature that would cause existing functionality to change)
- ✅ Documentation (non-code change)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
* Fix #3684 - CLI Yubikey is skipped when Yubikey feature is not enabled

* Skip building testguibrowser if browser integration is disabled

I also added a "no feature" build in the Linux CI. Future issues when features are disabled will be caught within the CI process.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested locally and with the new CI build.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: tests/TestCli.cpp [hunk 1]
```diff
@@ -1791,6 +1791,7 @@ void TestCli::testInvalidDbFiles()
  */
 void TestCli::testYubiKeyOption()
 {
+#ifdef WITH_XC_YUBIKEY
     if (!YubiKey::instance()->init()) {
         QSKIP("Unable to connect to YubiKey");
     }
```
Comment (phoerious) [inline tests/TestCli.cpp]: Shouldn't this be handled transparently by our YubiKey test stub?

Comment (droidmonkey) [reply tests/TestCli.cpp]: Perhaps? This might be an anomaly with how tests are built in that case.

Comment (phoerious) [reply tests/TestCli.cpp]: We have this stub class. Either we should use it everywhere or just not compile YubiKey tests at all.

Comment (droidmonkey) [reply tests/TestCli.cpp]: I figured out the root issue and fixed a hidden bug along the way.

**Comment by droidmonkey** [comment]
Ready for re-review

**Comment by louib** [comment]
@droidmonkey build failing on Windows:
```
TestCli::testClip() QTestLib: This test case check ("(((clipboard->text()) == (QString(""))))") failed because the requested timeout (1500 ms) was too short, 1550 ms would have been sufficient this time.
```

**Comment by droidmonkey** [comment]
Random anomoly

**Pull request merged**